app/helpers/url_helper.rb

```diff
@@ -0,0 +1,12 @@
+module UrlHelper
+  # Modifies the uri params.
+  # uri - URI object
+  # params - Hash, flat object e.g. { "page[per]" => 2 }
+  def self.add_params(uri, params)
+    params = Hash[URI.decode_www_form(uri.query || "") + params.to_a]
+    uri.query = params.map { |(key, value)|
```

Comment:
Could swap this to `URI.encode_www_form`, but I prefer `page[per]=1` instead of `page%5Bper%5D%3D1`